tests/foreman/ui/test_host.py

```diff
@@ -880,6 +880,49 @@ def test_negative_delete_primary_interface(session, module_host_template):
         assert 'Interface Delete button is disabled' in str(context.value)
 
 
+@tier2
+def test_positive_view_hosts_with_non_admin_user(test_name, module_org, module_loc):
+    """View hosts and content hosts as a non-admin user with only view_hosts, edit_hosts
+    and view_organization permissions
+
+    :BZ: 1642076
+
+    :id: 19a07026-0550-11ea-bfdc-98fa9b6ecd5a
+
+    :expectedresults: user with only view_hosts, edit_hosts and view_organization permissions
+        is able to read content hosts and hosts
+
+    :CaseLevel: System
+    """
+    user_password = gen_string('alpha')
+    role = entities.Role(organization=[module_org]).create()
+    create_role_permissions(
+        role,
+        {
+            'Organization': ['view_organizations'],
+            'Host': ['view_hosts'],
+        }
+    )
+    user = entities.User(
+        role=[role],
+        admin=False,
+        password=user_password,
+        organization=[module_org],
+        location=[module_loc],
+        default_organization=module_org,
+        default_location=module_loc,
+    ).create()
+    host = entities.Host(
+        location=module_loc,
+        organization=module_org,
+    ).create()
+    with Session(test_name, user=user.login, password=user_password) as session:
+        hosts = session.host.read_all()
+        assert any([True for h in hosts['table'] if h['Name'] == host.name])
```

Comment:
This feels unnecessary complicated. You filter the list on the fly and return True for every element that exists in new list.
Instead, you can just get the list of boolean values that tell if condition is true.
With generator comprehension, you can stop execution on first match.
```python
assert any(h['Name'] == host.name for h in hosts['table'])
```

Reply:
Instead writing only `h` please use `host`, helps in reading the code. Just a suggestion

Reply:
Thanks @mirekdlugosz.